### ISSUE TITLE
[a11y]: Dismissible tabs component issues

### ISSUE THREAD

**Issue by guidari** [carbon-design-system/carbon#14507]
### Package

@carbon/react

### Package version

v1.35.0

### Description

One minor accessibility issues with dismissible tabs.
- labelling

Labeling on dismissible tabs needs to be revamped.
For a mouse user, there are multiple targets inside each tab and they have separate names.
![Screenshot 2023-07-25 at 1.32.40 PM.png](https://zenhub.ibm.com/images/5cab73f442b8dc5ede7b84cc/b7f077ed-d5b4-4fa4-b17d-73a312d686ff)
For keyboard, there is only one focus point, so the two labels are concatenated into one message that doesn't make a lot of sense to a screen reader (e.g., "[tab name] close tab")

![Screenshot 2023-07-25 at 1.33.04 PM.png](https://zenhub.ibm.com/images/5cab73f442b8dc5ede7b84cc/c458745a-c270-4804-bae6-0d52f547c541)

A better description would be added with aria-describedby, so that the user heard "tab label, press Esc to dismiss'
BTW, we've previously talked about giving these examples 'real' names instead of "tab label".

### WCAG 2.1 Violation

Name, Role, Value

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-tabs--dismissable

### Steps to reproduce

Labelling
1. Open https://react.carbondesignsystem.com/?path=/story/components-tabs--dismissable
2. With mouse, hover over the x of the dismissible tabs to confirm they are called "Close tab"
3. With keyboard, tab into tablist and confirm that there is only one tabstop (i.e., the x is not not separately reachable by keyboard)
4. With Inspect, confirm that the name of the tabitem is a concatenation of the name of the tabitem and the name of the x

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
> BTW, we've previously talked about giving these examples 'real' names instead of "tab label".

I agree, lets do it. There's some examples of what we could use on the [usage page](https://carbondesignsystem.com/components/tabs/usage/), "IP Firewall", "User agent blocking", etc.
